### ISSUE TITLE
Ensure register/login links and views work correctly

### ISSUE THREAD

**Issue by brylie** [CiviWiki/OpenCiviWiki#918]
We need to make sure our "register" and "login" pages work correctly.

## Context
We are in the process of converting all of our templates to regular Django templates. Previously, the templates were created using Backbone.js and some other templating language.

We have also recently configured a custom user model.

## Bug report
@Floyd-Droid reports
> When I go to register (same info as superuser), I get two errors:
> (1) "Sorry, this username already exists"
> (2) "An account exists with this email address"
>
> This suggests the account already exists, but when I attempt to log in, I get a "Not found" error.
>
> If I try to register another user with different name and email, I have to click the register button twice before getting the same 2 messages above. When I try to log in with the new user info, I'm taken to the /beta page. Any help will be greatly appreciated.

## Task
- [ ] ensure that the "register" and "login" links point to the correct view
- [ ] ensure we use the custom `User` model for "register and "login" views
- [ ] ensure that the "register" and "login" pages use all styles from previous versions

## Reference
Split from discussion in issue #909

**Comment by brylie** [comment]
@a7madnassar would you be able to double-check that the views, templates, and related links for "register" and "login" all work correctly?

**Comment by a7madnassar** [comment]
@brylie yes, I will take a look at it. Just wondering if this bug is in the current authentication/registration or the new one? I will check both anyways.

**Comment by a7madnassar** [comment]
@brylie The problem is that the admin user is created using Django and the other users are created using the frontend. The app currently creates a User record as well as an Account record. The account record does not exist for the superuser because Django didn't create it. So, is it essential for superusers to access from the frontend? Django provides its own admin app and the superuser will be able to log in there fine.

If superusers need to access the app like other regular users. We have to do the following:
- Customize the process of creating superusers. The user can be registered as any other user, this will ensure the creation of an Account record for the user. Then a script can turn on the `is_superuser` flag on that user.

**Comment by brylie** [comment]
@a7madnassar we should use a single user model, which you created. If it hasn't already, Django needs to be configured to use that "default" user model throughout the project. As part of our migration to using Django templates, we should replace the login and registration views with class-based views that use the correct user model. Most of these pieces are already in place, we just need to follow up to make sure the custom user model is used throughout the app. Would you consider updating our templates to use the correct login and register class-based views that are based on the custom user model?

**Comment by brylie** [comment]
@shashankks0987 has also confirmed that our default user model is configured correctly. So, it seems like the remaining task is to just use the correct user model in our register/login views

**Comment by brylie** [comment]
@shashankks0987 and @a7madnassar would you mind collaborating on this issue? Our register/login views are critical for continued development of the project.

**Comment by a7madnassar** [comment]
@brylie and @shashankks0987 the code has been updated to reference the new custom user. That is not the issue. The current issues are:

1. A superuser cannot log in to the app. This is expected behavior because the current logic requires an `Account` record associated with the user. The superuser created by Django does not have an `Account` record. That's why the user cannot log in. In my opinion, superusers are meant for the site admin area, which they will be able to log to fine.
2. The registration process will hang for a long time if Celery workers are not running. This is caused by a Celery task to send the activation email.
I'm looking into a solution for no. 2

**Comment by a7madnassar** [comment]
@brylie and @shashankks0987 I came up with a solution to solve the registration issue. Currently, the code runs a Celery task to send the activation email on registration. I think having Celery as a dependency is not a great choice at this stage. Celery needs workers and a broker configured and running. This blocks completing the registration process correctly. I think Celery can be added eventually as the project grows.

My solution is to switch to synchronous email sending using Django's email functionality. In my setup, I don't have an email server running, so it hangs for a couple of seconds. In production, this shouldn't be a problem if an email server is configured and running. I implemented this in a way where if the `DEBUG` env variable is set by developers, the email sending functionality is bypassed. This solution allows the registration process to complete normally.

Let me know if you like this solution, and assign this issue to me and I can submit a pull request.

**Comment by brylie** [comment]
@a7madnassar great proposal. I'll make a separate task to track and discuss removing the Celery dependency. Would you mind if I assign you to the Celery removal task?

**Comment by brylie** [comment]
@a7madnassar are the login/register views using the correct `User` model?

**Comment by shashankks0987** [comment]
@brylie I'm being redirected to /beta page upon registration of a new user. Is this issue looking to fix the same?  
or is it trying to fix this:
```
(1) "Sorry, this username already exists"
(2) "An account exists with this email address"
```

I feel this issue is trying to solve two different problems

**Comment by brylie** [comment]
@shashankks0987 we should remove all references to "beta". There is an open issue to remove all "closed beta" code:
https://github.com/CiviWiki/OpenCiviWiki/issues/849

**Comment by a7madnassar** [comment]
> @a7madnassar are the login/register views using the correct `User` model?

@brylie yes, the code is using the new custom User.

**Comment by brylie** [comment]
OK, can you test out the login and register views please to make sure they are correct? Is the only login/register bug related to Celery?

**Comment by a7madnassar** [comment]
@brylie I tested the login/register functionality and once we remove the Celery dependency. Registration works correctly. As for login, only users created using the frontend can log in. I think this issue should be closed once Celery removal is done.

**Comment by a7madnassar** [comment]
@brylie I think we should close this issue now. Login, registration and password reset are all working correctly now.
